cmd/spire-agent/cli/run/run.go

```diff
@@ -356,6 +356,23 @@ func NewAgentConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool)
 	serverHostPort := net.JoinHostPort(c.Agent.ServerAddress, strconv.Itoa(c.Agent.ServerPort))
 	ac.ServerAddress = fmt.Sprintf("dns:///%s", serverHostPort)
 
+	logOptions = append(logOptions,
+		log.WithLevel(c.Agent.LogLevel),
+		log.WithFormat(c.Agent.LogFormat),
+		log.WithOutputFile(c.Agent.LogFile))
+
+	logger, err := log.NewLogger(logOptions...)
+	if err != nil {
+		return nil, fmt.Errorf("could not start logger: %s", err)
+	}
+	ac.Log = logger
+
+	// Warn on a non-conforming trust domain to avoid breaking backwards compatibility
+	if len(c.Agent.TrustDomain) > common_cli.MaxTrustDomainLength {
+		logger.Warnf("Configured trust domain should be less than %d characters to be SPIFFE compliant",
+			common_cli.MaxTrustDomainLength)
```

Comment:
I've been thinking about this over the weekend and had a discussion with @evan2645. Since there is no easy way to migrate to a new trust domain today, I hesitate to scare somebody enough that they feel they need to do it. I wonder if we can be a little more clear of the implication of using a longer trust domain so folks can make an educated choice.

Maybe something like: 
```suggestion
		logger.Warnf("Configured trust domain name should be less than %d characters to be SPIFFE compliant; a longer trust domain name may impact interoperability",
			common_cli.MaxTrustDomainLength)
```

Reply:
may we add TD as a field? to make easier to user to know what td was configured (and dont have to move to config files.